tests/core/pyspec/eth2spec/test/electra/block_processing/test_process_withdrawals.py

```diff
@@ -94,14 +94,351 @@ def test_pending_withdrawals_one_skipped_one_effective(spec, state):
+        pending_withdrawal_requests=pending_withdrawal_requests[:spec.MAX_PENDING_PARTIALS_PER_WITHDRAWALS_SWEEP]
+    )
+
+    withdrawals_exceeding_max = pending_withdrawal_requests[spec.MAX_PENDING_PARTIALS_PER_WITHDRAWALS_SWEEP:]
+    assert state.pending_partial_withdrawals == withdrawals_exceeding_max
+
+
+@with_electra_and_later
+@spec_state_test
+def test_pending_withdrawals_exiting_validator(spec, state):
+    validator_index = len(state.validators) // 2
+
+    pending_withdrawal = prepare_pending_withdrawal(spec, state, validator_index)
+    spec.initiate_validator_exit(state, pending_withdrawal.index)
+
+    execution_payload = build_empty_execution_payload(spec, state)
+    yield from run_withdrawals_processing(spec, state, execution_payload, num_expected_withdrawals=0)
+
+    assert state.pending_partial_withdrawals == []
+
+
+@with_electra_and_later
+@spec_state_test
+def test_pending_withdrawals_low_effective_balance(spec, state):
+    validator_index = len(state.validators) // 2
+
+    pending_withdrawal = prepare_pending_withdrawal(spec, state, validator_index)
+    state.validators[pending_withdrawal.index].effective_balance = (
+        spec.MIN_ACTIVATION_BALANCE - spec.EFFECTIVE_BALANCE_INCREMENT
+    )
+
+    execution_payload = build_empty_execution_payload(spec, state)
+    yield from run_withdrawals_processing(spec, state, execution_payload, num_expected_withdrawals=0)
+
+    assert state.pending_partial_withdrawals == []
+
+
+@with_electra_and_later
+@spec_state_test
+def test_pending_withdrawals_no_excess_balance(spec, state):
+    validator_index = len(state.validators) // 2
+
+    pending_withdrawal = prepare_pending_withdrawal(spec, state, validator_index)
+    state.balances[pending_withdrawal.index] = spec.MIN_ACTIVATION_BALANCE
+
+    execution_payload = build_empty_execution_payload(spec, state)
+    yield from run_withdrawals_processing(spec, state, execution_payload, num_expected_withdrawals=0)
+
+    assert state.pending_partial_withdrawals == []
+
+
+@with_electra_and_later
+@spec_state_test
+def test_pending_withdrawals_with_ineffective_sweep_on_top(spec, state):
+    # Ensure validator will be processed by the sweep
+    validator_index = min(len(state.validators), spec.MAX_VALIDATORS_PER_WITHDRAWALS_SWEEP) // 2
+
+    pending_withdrawal = prepare_pending_withdrawal(
+        spec, state,
+        validator_index,
+        effective_balance=spec.MAX_EFFECTIVE_BALANCE_ELECTRA,
+    )
+
+    # Check that validator is partially withdrawable before pending withdrawal is processed
+    assert spec.is_partially_withdrawable_validator(
+        state.validators[validator_index],
+        state.balances[validator_index]
+    )
+    # And is not partially withdrawable thereafter
+    assert not spec.is_partially_withdrawable_validator(
+        state.validators[validator_index],
+        state.balances[validator_index] - pending_withdrawal.amount
+    )
```

Comment:
I think it would make more sense if the second assert were after `run_withdrawals_processing`. Without the `pending_withdrawal.amount` subtraction of course. But if this is for consistency with the other sweep-on-top tests, I suppose that's fine.

Reply:
It is not for consistency but to ensure that the test case is set up correctly. Do you think it doesn’t make sense?

Reply:
It makes sense. But it kind of assumes the withdrawal works as expected. Should be okay though.